src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -392,7 +416,11 @@ private boolean isLikelyNamespace(JSType type, Compiler compiler) {
     if (utype != null) {
       return isNativeObjectType(utype.restrictByNotNullOrUndefined(), compiler);
     }
-    return isNativeObjectType(type, compiler);
+    // The inferred type for an namespace-like object is not the native object,
+    // but rather PrototypeObjectType.
+    return isNativeObjectType(type, compiler) ||
+        // TODO(rado): replace with something shorter that makes all tests pass.
```

Comment:
I give up, can't find a better API to check for PrototypeObjectType. But I want to leave a comment to show my aspirations for a better api :) Changed the comment a bit.